src/go/wsl-helper/pkg/dockerproxy/serve.go

```diff
@@ -79,7 +79,7 @@ func Serve(endpoint string, dialer func() (net.Conn, error)) error {
 	munger := newRequestMunger()
 	proxy := &util.ReverseProxy{
 		Dial: func(string, string) (net.Conn, error) {
-			return dialer()
+			return dialer(ctx)
 		},
```

Comment:
`util.ReverseProxy` should take a `DialContext` instead, because it has a context from the request to be forwarded.